src/_guides/openapi/design-first-spring.md

```diff
@@ -0,0 +1,388 @@
+the documentation first, you can see if people like the look of the API before
+you waste time building it.
+
+Then, as you progress through, especially if you add these tools to an existing
+codebase, you will continue to find mistakes in your OpenAPI or your actual API
+code. Improve both as you go until you have a perfect match that will never
+again be broken, solving the "docs vs code" drift problem, and every fix will be
+deployed to Bump.sh with each commit/merge.
+
+```bash
+$ bump deploy openapi.yaml \
+  --doc spring-design-first \
+  --token my-documentation-token
+
+* Your new documentation version will soon be ready at https://bump.sh/hub/code-samples/doc/spring-design-first
+```
+
+Instead of using the [CLI](https://github.com/bump-sh/cli#bump-deploy-file) you
+could use [GitHub
+Actions](https://github.com/marketplace/actions/bump-sh-api-documentation-changelog),
+or a bunch of other [Continuous Integration](/help/continuous-integration/).
+
+Once Bump.sh is hooked up, let's look at how we'd teach a Spring Boot API (new,
+or existing) to handle request validation for us.
+
+## Request Validation with OpenAPI in Spring Boot
+
+In a Spring Boot application, you can use OpenAPI to validate incoming requests
+against the API description. This ensures that the requests conform to the
+expected structure and data types defined in the OpenAPI specification.
+
+To demonstrate this, let's assume we have a simple Spring Boot application with
+an `Employee` entity that has two properties: `name` and `role`. We'll create a
+controller that handles incoming requests to create new employees, but it
+doesn't do any validation just yet.
+
+```java
+@RestController
+class EmployeeController {
+
+	private final EmployeeRepository repository;
+
+	EmployeeController(EmployeeRepository repository) {
+		this.repository = repository;
+	}
+
+	@GetMapping("/employees")
+	List<Employee> all() {
+		return repository.findAll();
+	}
+
+	@PostMapping("/employees")
+	Employee newEmployee(@RequestBody Employee newEmployee) {
+		return repository.save(newEmployee);
+	}
+```
+
+When a request is made it will try and create a record with `name` and `role`,
+but if either property is missing (or misspelled) it will just save it anyway. 
+
+```bash
+curl -iXPOST -H 'content-type:application/json' \
+  localhost:8080/employees \
+  --data '{"role":"thief","nam":"Bilbo"}'
+```
+
+```http
+HTTP/1.1 200
+Content-Type: application/json
+
+{"id":3,"name":null,"role":"thief"}
+```
+
+Oops!
+
+This API needs more validation.
+
+**Bringing OpenAPI into the Spring Boot Application**
+
+If we're going to teach Spring Boot and OpenAPI to get along, let's first move OpenAPI somewhere Spring Boot is expecting it and de-clutter our root in the process. A great place for this is the `src/main/resources/api/` directory.
+
+```shell
+mkdir -p src/main/resources/api
+mv openapi.yaml src/main/resources/api/openapi.yaml
```

Comment:
nit: would it change a lot to move this under `.../resources/openapi/openapi.yaml` ? if the relative path contains an `openapi` directory then kappa will consider it as a "root dircetory", so it will omit the `file:/Users/...` prefix from the schema locations. Overall it makes the outputs nicer.

Reply:
> Is that just renaming the api to openapi

Yes, that's it :)

Reply:
Hello @philsturgeon Kappa right now doesn't understand wildcard content types (`*/*`) which are used in the 404 response definitions of openapi.yaml. Please update them to `application/json` for now - I will try to find some time in August to fix it, but it is unlikely to happen sooner.

Reply:
> you said about switching to openapi/ would filter file:/Users out of the output doesn't seem to be happening.

Indeed. Actually it is done only by the [default ValidationFailureSender](https://github.com/erosb/kappa/blob/master/kappa-servlet-adapter/src/main/java/com/github/erosb/kappa/operation/validator/adapters/server/servlet/ValidationFailureSender.java#L41) but that's not used by this example.

What do you think about:
 * removing the `RFC9457FailureSender` from this example project, moving it over to Kappa
 * fix it to exclude the `file:/Users` part from `"schemaLocation"` like the default sender does
 * release a new kappa version and use it in this example?

Reply:
I would like all of that very much please! :D